app/controllers/public/cart_products_controller.rb

```diff
@@ -13,12 +13,21 @@ def create
   end
 
   def update
+    cart_product = CartProduct.find(params[:id])
+    cart_product.update(cart_product_params)
```

Comment:
ログイン中の顧客IDと対象のカートアイテムの顧客IDが一致するかチェックした方が安全だと思います。
```suggestion
    if cart_product.customer_id == current_customer.id
      cart_product.update(cart_product_params)
    end
```